pkg/txn/storage/tae/storage.go

```diff
@@ -26,71 +34,169 @@ import (
+// Close implements storage.TxnTAEStorage
+func (s *TAEStorage) Close(ctx context.Context) error {
 	panic("unimplemented")
 }
 
-// Committing implements storage.TxnStorage
-func (*Storage) Committing(ctx context.Context, txnMeta txn.TxnMeta) error {
-	panic("unimplemented")
+// Commit implements storage.TxnTAEStorage
+func (s *TAEStorage) Commit(ctx context.Context, txnMeta txn.TxnMeta) error {
+	txnHandle, err := s.get(txnMeta)
+	if err != nil {
+		return err
+	}
+	return txnHandle.Commit()
+
 }
 
-// Destroy implements storage.TxnStorage
-func (*Storage) Destroy(ctx context.Context) error {
+// Committing implements storage.TxnTAEStorage
+func (s *TAEStorage) Committing(ctx context.Context, txnMeta txn.TxnMeta) error {
 	panic("unimplemented")
 }
 
-// Prepare implements storage.TxnStorage
-func (*Storage) Prepare(ctx context.Context, txnMeta txn.TxnMeta) (timestamp.Timestamp, error) {
+// Destroy implements storage.TxnTAEStorage
+func (s *TAEStorage) Destroy(ctx context.Context) error {
 	panic("unimplemented")
 }
 
-// Read implements storage.TxnStorage
-func (*Storage) Read(ctx context.Context, txnMeta txn.TxnMeta, op uint32, payload []byte) (storage.ReadResult, error) {
+// Prepare implements storage.TxnTAEStorage
+func (s *TAEStorage) Prepare(ctx context.Context, txnMeta txn.TxnMeta) (timestamp.Timestamp, error) {
 	panic("unimplemented")
 }
 
-// Rollback implements storage.TxnStorage
-func (*Storage) Rollback(ctx context.Context, txnMeta txn.TxnMeta) error {
+// Read implements storage.TxnTAEStorage
+func (s *TAEStorage) Read(ctx context.Context, txnMeta txn.TxnMeta, op uint32, payload []byte) (storage.ReadResult, error) {
 	panic("unimplemented")
 }
 
-// StartRecovery implements storage.TxnStorage
-func (*Storage) StartRecovery(context.Context, chan txn.TxnMeta) {
-	panic("unimplemented")
+// Rollback implements storage.TxnTAEStorage
+func (s *TAEStorage) Rollback(ctx context.Context, txnMeta txn.TxnMeta) error {
+	txnHandle, err := s.get(txnMeta)
+	if err != nil {
+		return err
+	}
+	return txnHandle.Rollback()
 }
 
-// Write implements storage.TxnStorage
-func (*Storage) Write(ctx context.Context, txnMeta txn.TxnMeta, op uint32, payload []byte) ([]byte, error) {
+// StartRecovery implements storage.TxnTAEStorage
+func (s *TAEStorage) StartRecovery(context.Context, chan txn.TxnMeta) {
 	panic("unimplemented")
 }
+
+// Write implements storage.TxnTAEStorage
+func (s *TAEStorage) Write(ctx context.Context, txnMeta txn.TxnMeta, op uint32, payload []byte) ([]byte, error) {
+	txnHandle, err := s.getOrCreate(txnMeta)
+	if err != nil {
+		return nil, err
+	}
+
+	precommitW := &api.PrecommitWriteCmd{}
+	err = types.Decode(payload, precommitW)
+	if err != nil {
+		return nil, err
+	}
+	for _, entry := range precommitW.EntryList {
+		//err = doHandleCmd(txnHandle, entry)
+		err = txnHandle.HandleCmd(entry)
+	}
+	return nil, err
+}
+
+//func doHandleCmd(txn txnif.AsyncTxn, cmd *api.Entry) (err error) {
```

Comment:
The commented code needs to be removed directly

Reply:
ok